scripts/valhalla_build_config

```diff
@@ -588,8 +588,19 @@ def override_config(args_: dict, leaves_: list, config_: dict):
         for k in keys[:-1]:
             v = v[k]
         v[keys[-1]] = args_.get(leaf.replace('\0', '_'))
-        if isinstance(v[keys[-1]], type(Optional())):
-            del v[keys[-1]]
+        sanitize_config(v)
```

Comment:
this doesnt seem quite right, we replaced a single `if` with a recursive function. i think perhaps my earlier suggestion didnt make any sense! i had forgotten that of course we already do remove the optionals since they arent actually holding usable values. now i guess im wondering, can we just call `override_config` as is from the new improved bindings? sorry im a bit dense at the moment!

Reply:
hehe, yeah the suggestion was borderline, I was about to say smth, then I thought like "well 2 lines saved, just do it". the code is "correct" in that it does the right thing, but I agree, it doesn't "seem" right:)

we can't just call `override_config` as-is, it's expecting lots of args like the config string with `\0` termination and so on. I'll pull that function again to the other file as it was before, don't really feel like touching again the slightly brittle logic in valhalla_build_config just to save a 3-4 lines of code. ok for you?